### PR TITLE
ddl: bug in columnDefToCol, Length shouldn't be set for unspecified fields. (#2381)

### PR DESCRIPTION
**Pull request by idlesummerbreeze**
Hi @coocood,

In according to #2380 I create this PR.

Regards,
Yu

### CODE REVIEW

**Comment by idlesummerbreeze** [comment]
Hi @coocood ,

after correcting some small errors, this change pass all checks.

**Comment by shenli** [comment]
@idlesummerbreeze Well done!
Could you add a uint test in ddl package?

**Comment by coocood** [comment]
@idlesummerbreeze 
You can add a test that create a table with a unique column, then get IndexColumn via InfoSchema, check that it's IndexColumn length is -1.

**Comment by idlesummerbreeze** [comment]
Hi @all,

no problem, I will add a test!

Regards,
Yu

2017-01-04 2:58 GMT+01:00 Ewan Chou <notifications@github.com>:

File: util/types/etc.go [hunk 1]
```diff
@@ -49,6 +49,12 @@ func IsTypeChar(tp byte) bool {
 	}
 }
 
+//IsTypeSpecifiable returns a boolean indicating
+// whether the tp is a specifiable type
```
Comment (hanfei1991) [inline util/types/etc.go]: add punctuation `.` at end of comments.

**Pull request merged**